build/ublue-os-just/libfunctions.sh

```diff
@@ -0,0 +1,167 @@
+        # and remove "noconfirm" from $1 when assigning it to ACTION
+        ACTION="${1/noconfirm/}"
+    fi
+
+    # If a filename is provided
+    if [ -n "$2" ]; then
+        # Set FILE to the provided filename
+        FILE="$2"
+    fi
+
+    # If container name is ALL
+    if [ "$3" == "ALL" ] || [ -z "$3" ]; then
+        if [[ ! "$1" =~ ^noconfirm ]]; then
+            # Ask user if they REALLY want to assemble all the containers
+            echo -e "${b}WARNING${n}: This will assemble and ${u}replace${n}\nALL containers defined in ${b}$FILE${n}."
+            CONFIRM=$(Confirm "Are you sure you want to do this?")
+            if [ "$CONFIRM" == "1" ]; then
+                echo "Aborting..."
+                return 1
+            fi
+        fi
+        # Run the distrobox assemble command
+        distrobox assemble "$ACTION" --file "$FILE" --replace
+    else
+        # Set distrobox name to provided name
+        NAME="$3"
+    fi
+    
+    # If we do not want confirmations
+    if [[ ! "$1" =~ ^noconfirm ]]; then
+        # Ask the user if they really want to replace $NAME container
+        echo -e "${b}WARNING${n}: This will assemble and ${u}replace${n} the container ${b}$NAME${n}\nwith the one defined in ${b}$FILE${n}."
+        CONFIRM=$(Confirm "Are you sure you want to do this?")
+        if [ "$CONFIRM" == "1" ]; then
+            echo "Aborting..."
+            return 1
+        fi
+    fi
+
+    # Run the distrobox assemble command
+    distrobox assemble "$ACTION" --file "$FILE" --name "$NAME" --replace
+}
+
+########
+## Function to parse a distrobox.ini file and make a selectable list from it
+########
+## Parse a distrobox.ini manifest and let user select which container to setup
+# AssembleParse "$HOME/distrobox.ini" create
+## Parse a distrobox.ini manifest and create ubuntu container without confirmation
+# AssembleParse "$HOME/distrobox.ini" noconfirmcreate ubuntu
+function AssembleList (){
+    # Set defaults
+    FILE="$1"
+    ACTION="create"
+    CHOICE="prompt"
+
+    # If an ACTION is supplied
+    if [ -n "$2" ]; then
+        # Replace default action
+        ACTION="$2"
+    fi
+    
+    # If a CHOICE is predefined
+    if [ -n "$3" ]; then
+        # Replace default choice
+        CHOICE="$3"
+    fi
+
+    # If the choice is "prompt" then ask user what container they want
+    if [ "$CHOICE" == "prompt" ]; then
+        CONTAINERS=$(grep -P "\[.+\]" "$FILE" | sed -E 's/\[(.+)\]/\1/')
+        echo "${b}Pre-defined Containers${n}"
+        echo "Please select a container to create"
+        # Disable an irrelevant shellscheck for next line as we want word splitting
+        # shellcheck disable=SC2086
+        CHOICE=$(Choose ALL $CONTAINERS)
+    fi
+
+    # If choice is not empty by now (will be empty if escaped from Choice function)
+    if [ -n "$CHOICE" ]; then
+        # Assemble the selected container
+        Assemble "$ACTION" "$FILE" "$CHOICE"
+    fi
+}
```

Comment:
Seems like the `Distrobox`, `Assemble`, and `AssembleList` functions are all unused in this PR.

Looks like they are meant to be be part of #197 

Can we move them to that PR please?

Reply:
will do

Reply:
think i did it correctly, should be in #197 now